tuning/utils/data_utils.py

```diff
@@ -0,0 +1,34 @@
+import re
+from datasets import Dataset
+
+def formatting_function(dataset, template, eos_token=""):
+    """Function to format datasets with Alpaca style / other templates.
+    Args:
+        dataset: the HF Dataset element loaded from a JSON or DatasetDict object.
+        template: Template to format data with. Features of Dataset 
+            should be referred to by {{key}}
+    Returns:
+        Formatted HF Dataset, dataset_field name that contains formatted data.
+    """
+
+    formatted_dataset_field = "formatted_data_field"
+    template += eos_token
```

Comment:
Will this cause issues if the template provided already has a terminating EOS token?

Reply:
apparently not, I just duplicated functionality already in fms-tuning https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tuning/sft_trainer.py#L235 where a formatter with EOS token is applied. I just didnt want us to loop over dataset twice hence moved that here . Quality has been tested with existing behavior.